### PR TITLE
Issue 37592: Add search document for SampleSets

### PR DESCRIPTION
**Pull request by labkey-ians**
https://www.labkey.org/home/Developer/issues/issues-details.view?issueId=37592

### CODE REVIEW

File: experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java [hunk 1]
```diff
@@ -569,4 +592,70 @@ public String toString()
...
+                    url.setExtraPath(getContainer().getId());
+
+                    Map<String, Object> props = new HashMap<>();
+                    Set<String> identifiersHi = new HashSet<>();
+
+                    // Name is identifier with highest weight
+                    identifiersHi.add(getName());
+
+                    // Add aliases in parenthesis in the title
+                    StringBuilder title = new StringBuilder("Sample source - " + getName());
```
Comment (labkey-susanh) [inline experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java]: Can we change this title to be just the name?

Comment (labkey-susanh) [reply experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java]: Actually, I think I'll change my mind here.  Using "Sample Set - <name>" will make sense for search results on LabKey Server side.  If there is a description, will that show up as the summary?

File: experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java [hunk 2]
```diff
@@ -569,4 +592,70 @@ public String toString()
     {
         return "SampleSet " + getName() + " in " + getContainer().getPath();
     }
+
+    public void index(SearchService.IndexTask task)
+    {
+        // Big hack to prevent study specimens from being indexed as
+        if (StudyService.SPECIMEN_NAMESPACE_PREFIX.equals(getLSIDNamespacePrefix()))
```
[the remark this answers is not in the document]
Comment (labkey-susanh) [reply experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java]: Is the REVIEW comment for now or for later?

File: experiment/resources/schemas/exp.xml [hunk 3]
```diff
@@ -368,7 +368,10 @@
         <description>Contains the user that last modified this sample set</description>
       </column>
       <column columnName="Container">
-          <description>Contains the folder or project in which this sample set is stored</description>
+        <description>Contains the folder or project in which this sample set is stored</description>
+      </column>
+      <column columnName="LastIndexed">
+        <description>Contains the time this material source was last updated in the search index</description>
```
Comment (labkey-susanh) [inline experiment/resources/schemas/exp.xml]: This field should probably not be editable.  So we should set the appropriate properties here.

File: experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (labkey-ians) [reply experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java]: both, It isn't clear to me what that check is preventing, just carried it over from the individual sample index method.

File: experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java [hunk 4]
```diff
@@ -569,4 +592,70 @@ public String toString()
...
+        }
+        if (task == null)
+        {
+            SearchService ss = SearchService.get();
+            if (null == ss)
+                return;
+            task = ss.defaultTask();
+        }
+
+        // do the least possible amount of work here
```
[the remark this answers is not in the document]
Comment (labkey-ians) [reply experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java]: removed.

File: experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java [hunk 5]
```diff
@@ -569,4 +592,70 @@ public String toString()
...
+
+                    // Name is identifier with highest weight
+                    identifiersHi.add(getName());
+
+                    // Add aliases in parenthesis in the title
+                    StringBuilder title = new StringBuilder("Sample source - " + getName());
+
+                    props.put(SearchService.PROPERTY.categories.toString(), searchCategory.toString());
+                    props.put(SearchService.PROPERTY.title.toString(), title.toString());
+                    props.put(SearchService.PROPERTY.keywordsLo.toString(), "SampleSet");      // Treat the word "Sample" a low priority keyword
```
[the remark this answers is not in the document]
Comment (labkey-ians) [reply experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java]: fixed

File: experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java (hunk 1 above)
Comment (labkey-ians) [reply experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java]: Updated text.

File: experiment/resources/schemas/exp.xml (hunk 3 above)
Comment (labkey-ians) [reply experiment/resources/schemas/exp.xml]: fixed with newest commits.

File: experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (labkey-ians) [reply experiment/src/org/labkey/experiment/api/ExpSampleSetImpl.java]: Removed review tag, and expanded comment clarifying need for if clause.

**Pull request merged**